### ISSUE TITLE
BookmarksPanel crashes changing device orientation

### ISSUE THREAD

**Issue by garvankeeley** [mozilla-mobile/firefox-ios#6722]
STR unknown, from Xcode crash reporter:

```
Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Triggered by Thread:  0

Last Exception Backtrace:
0   CoreFoundation                	0x1938b6a48 __exceptionPreprocess + 220 (NSException.m:199)
1   libobjc.A.dylib               	0x1935ddfa4 objc_exception_throw + 56 (objc-exception.mm:565)
2   CoreFoundation                	0x1937b8e88 +[NSException raise:format:arguments:] + 96 (NSException.m:146)
3   Foundation                    	0x193bee654 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 128 (NSException.m:231)
4   UIKitCore                     	0x197aea824 -[UITableView _endCellAnimationsWithContext:] + 6076 (UITableView.m:2161)
5   UIKitCore                     	0x197b0479c -[UITableView _updateRowsAtIndexPaths:withUpdateAction:rowAnimation:usingPresentationValues:] + 624 (UITableView.m:7595)
6   UIKitCore                     	0x197b04d50 -[UITableView reloadRowsAtIndexPaths:withRowAnimation:] + 212 (UITableView.m:7618)
7   Client                        	0x10223e1a0 SiteTableViewController.applyTheme() + 1380 (SiteTableViewController.swift:202)
8   Client                        	0x10238c250 BookmarksPanel.applyTheme() + 20 (BookmarksPanel.swift:158)
9   Client                        	0x102274600 LibraryViewController.applyTheme() + 396 (LibraryViewController.swift:205)
10  Client                        	0x1023300dc BrowserViewController.applyTheme() + 900 (BrowserViewController.swift:2426)
11  Client                        	0x102330950 @objc BrowserViewController.displayThemeChanged(notification:) + 108 (BrowserViewController.swift:190)
12  CoreFoundation                	0x193812288 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 20 (CFNotificationCenter.c:787)
13  CoreFoundation                	0x1938122d0 ___CFXRegistrationPost1_block_invoke + 64 (CFNotificationCenter.c:175)
14  CoreFoundation                	0x193811630 _CFXRegistrationPost1 + 368 (CFNotificationCenter.c:198)
15  CoreFoundation                	0x1938112e8 ___CFXNotificationPost_block_invoke + 104 (CFNotificationCenter.c:1371)
16  CoreFoundation                	0x19378d59c -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1416 (CFXNotificationRegistrarOld.m:168)
17  CoreFoundation                	0x193810c38 _CFXNotificationPost + 1244 (CFNotificationCenter.c:1359)
18  Foundation                    	0x193b6d0c8 -[NSNotificationCenter postNotificationName:object:userInfo:] + 60 (NSNotification.m:576)
19  Client                        	0x1021321cc ThemeManager.current.didset + 356 (ThemeManager.swift:19)
20  Client                        	0x1021322e8 ThemeManager.current.setter + 116 (<compiler-generated>:0)
21  Client                        	0x10233ca74 BrowserViewController.traitCollectionDidChange(_:) + 284 (BrowserViewController.swift:1160)
22  Client                        	0x10233cad4 @objc BrowserViewController.traitCollectionDidChange(_:) + 52 (<compiler-generated>:0)
```

**Comment by garvankeeley** [comment]
Hard to repro, I was unable to do so. Is a top crasher as reported in Xcode.

**Comment by SimonBasca** [comment]
I think I manage to reproduce this issue by following the STR: 
1. Open Firefox 
2. Go Library - > Bookmarks
3. Background Firefox 
4. Go to iOS Settings -> Display & Text Size -> Enable "Bold Text"
5. Foreground Firefox. 

@garvankeeley I haven't been able to extract a crash log from this device. Can you please check if this is the actual crash?

**Comment by garvankeeley** [comment]
Nice find!! This looks like the crash yes.

**Comment by SimonBasca** [comment]
Verifying as fix on main 89b961b. The crash no longer occurs.
